### PR TITLE
[SPARK-13613] [ML] Provide ignored tests to export test dataset into CSV format

### PR DESCRIPTION
**Pull request by yanboliang**
## What changes were proposed in this pull request?

Provide ignored test cases to export the test dataset into CSV format in `LinearRegressionSuite`, `LogisticRegressionSuite`, `AFTSurvivalRegressionSuite` and `GeneralizedLinearRegressionSuite`, so users can validate the training accuracy compared with R's glm, glmnet and survival package.
cc @mengxr 
## How was this patch tested?

The test suite is ignored, but I have enabled all these cases offline and it works as expected.

### CODE REVIEW

File: mllib/src/test/scala/org/apache/spark/ml/regression/LinearRegressionSuite.scala [hunk 1]
```diff
@@ -60,8 +47,8 @@ class LinearRegressionSuite
         intercept = 6.3, weights = Array(4.7, 7.2), xMean = Array(0.9, -1.3),
         xVariance = Array(0.7, 1.2), nPoints = 10000, seed, eps = 0.1), 2))
     /*
-       datasetWithoutIntercept is not needed for correctness testing but is useful for illustrating
-       training model without intercept
+       datasetWithDenseFeatureWithoutIntercept is not needed for correctness testing
```
Comment (yanboliang) [inline mllib/src/test/scala/org/apache/spark/ml/regression/LinearRegressionSuite.scala]: Fix typo BTW.

**Comment by yanboliang** [comment]
ping @mengxr

File: mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala [hunk 2]
```diff
@@ -65,12 +51,23 @@ class LogisticRegressionSuite
...
   }
 
+  /**
+   * Enable the ignored test to export the dataset into CSV format,
+   * so we can validate the training accuracy compared with R's glmnet package.
+   */
+  ignore("export test data into CSV format") {
+    binaryDataset.rdd.map { case Row(label: Double, features: Vector) =>
+      label + "," + features.toArray.mkString(",")
+    }.repartition(1).saveAsTextFile("path")
```
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala]: Do not use `path`. We can try `target/tmp/LogisticRegressionSuite/binaryDataset`.

File: mllib/src/test/scala/org/apache/spark/ml/regression/AFTSurvivalRegressionSuite.scala [hunk 3]
```diff
@@ -44,6 +44,19 @@ class AFTSurvivalRegressionSuite
...
   }
 
+  /**
+   * Enable the ignored test to export the dataset into CSV format,
+   * so we can validate the training accuracy compared with R's survival package.
+   */
+  ignore("export test data into CSV format") {
+    datasetUnivariate.rdd.map { case Row(features: Vector, label: Double, censor: Double) =>
+      features.toArray.mkString(",") + "," + censor + "," + label
+    }.repartition(1).saveAsTextFile("univariate-path")
```
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/ml/regression/AFTSurvivalRegressionSuite.scala]: ditto

**Comment by mengxr** [comment]
LGTM except one minor issue on the output path, which should be in a temporary folder. So it doesn't appear in git.

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**